src/core/tsi/s2a/handshaker/s2a_tsi_handshaker.cc

```diff
@@ -78,15 +82,159 @@ typedef struct s2a_tsi_handshaker_result {
    *  client-side, and false iff the handshaker result is on the server-side.
    * **/
   bool is_client;
+  /** An open channel to the S2A handshaker service. **/
+  grpc_channel* channel;
 } s2a_tsi_handshaker_result;
 
+struct s2a_tsi_handshaker_continue_handshaker_next_args {
+  s2a_tsi_handshaker* handshaker;
+  std::unique_ptr<uint8_t> received_bytes;
+  size_t received_bytes_size;
+  tsi_handshaker_on_next_done_cb cb;
+  void* user_data;
+  grpc_closure closure;
+};
+
+/** A gRPC-provided callback function that is used when gRPC thread model is
+ * applied. */
+static void on_handshaker_service_resp_recv(void* arg, grpc_error* error) {
+  S2AHandshakerClient* client = static_cast<S2AHandshakerClient*>(arg);
+  if (client == nullptr) {
+    gpr_log(GPR_ERROR,
+            "The |S2AHandshakerClient| is nullptr in "
+            "|on_handshaker_service_resp_recv|.");
+    return;
+  }
+  bool success = true;
+  if (error != GRPC_ERROR_NONE) {
+    gpr_log(GPR_ERROR,
+            "In the S2A TSI handshaker's |on_handshaker_service_resp_recv| "
+            "method, there is the error: %s",
+            grpc_error_string(error));
+    success = false;
+  }
+  client->HandleResponse(success);
+}
+
+static tsi_result s2a_tsi_handshaker_continue_handshaker_next(
```

Comment:
Could you please explain what this function does? It is not clear from the API signature. It is also confusing with `tsi_handshaker_next()`.